dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/icon/IconController.java

```diff
@@ -126,17 +128,58 @@ public void getIconData(@PathVariable String key, HttpServletResponse response)
   @GetMapping
   public @ResponseBody List<IconResponse> getAllIcons(
       @RequestParam(required = false) String[] keywords) {
+
+    IconCriteria iconCriteria = IconCriteria.of(0, 0, true);
```

Comment:
I can understand the idea here but it is quite difficult to deduce the behavior from the code.
Yu could create a static constructor in `IconCriteria` named `SkipPagination` or something similar